mlflow/models/model.py

```diff
@@ -1012,20 +1027,17 @@ def set_model(model):
     """
     from mlflow.pyfunc import PythonModel
 
+    if isinstance(model, str):
+        raise mlflow.MlflowException(SET_MODEL_ERROR)
+
     if not (isinstance(model, PythonModel) or callable(model)):
-        try:
-            from mlflow.langchain import _validate_and_prepare_lc_model_or_path
-
-            # If its not a PyFuncModel, then it should be a Langchain model (not a path)
-            # Check this since the validation function does not
-            if isinstance(model, str):
-                raise mlflow.MlflowException(
-                    "Model should either be an instance of PyFuncModel or Langchain type."
-                )
-            model = _validate_and_prepare_lc_model_or_path(model, None)
-        except Exception as e:
-            raise mlflow.MlflowException(
-                "Model should either be an instance of PyFuncModel or Langchain type."
-            ) from e
+        for validate_function in [_validate_langchain_model, _validate_llama_index_model]:
+            try:
+                globals()["__mlflow_model__"] = validate_function(model)
+                return
+            except Exception:
+                pass
+        else:
+            raise mlflow.MlflowException(SET_MODEL_ERROR)
 
     globals()["__mlflow_model__"] = model
```

Comment:
is this line still necessary? since we either return or raise in the above loop

Reply:
This is for `PythonModel` / `callable`. Indeed, I think it's more natural to reverse the condition at L1033.